lib/event.rb

```diff
@@ -0,0 +1,43 @@
+class Event
+  attr_reader :name,
+              :crafts,
+              :attendees
+
+  def initialize(name, crafts, attendees)
+    @name = name
+    @crafts = crafts
+    @attendees = attendees
+  end
+
+  def attendee_names
+    @attendees.map do |attendee|
+      attendee.name
+    end
+  end
+
+  def craft_with_most_supplies
+    craft = @crafts.max_by do |craft|
+      craft.supplies_required.keys.count
+    end
+    craft.name
+  end
+
+  def supply_list
+    @crafts.map do |craft|
+      craft.supplies_required.keys.map do |key|
+        key.to_s
+      end
+    end.flatten.uniq
+  end
+
+  def attendees_by_craft_interest
+    new_hash = {}
+    keys = @crafts.find_all do |craft|
+      craft.name
+    end
+
+    keys.each do |key|
+      new_hash[key] = @attendees.find_all { |attendee| attendee.interests == key}
+    end
+  end
```

Comment:
A few mistakes here.

1. You aren't returning the `new_hash` at the end of the method. That means it will return `keys`.
2. You are using `find_all` which returns all items for a list where `craft.name` == true. You want to be using `.map` to create a new array and run `.name` on each craft.
3. `attendee.interests` is a list so we want to check if that list contains the key we are looking for not that it is equal to it
```suggestion
  def attendees_by_craft_interest
    new_hash = {}
    keys = @crafts.map do |craft|
      craft.name
    end

    keys.each do |key|
      new_hash[key] = @attendees.find_all { |attendee| attendee.interests.include?(key)}
    end
    new_hash
  end
```